ui/src/app/components/copy-modal.tsx

```diff
@@ -45,6 +45,10 @@ export interface Props {
   saveFunction: (CopyRequest) => Promise<FileDetail | ConceptSet>;
 }
 
+interface HocProps extends Props, NavigationProps {
```

Comment:
I'm not a huge fan of it either.. we won't need to use it when we migrate to hooks at least. I think explicitly declaring one per component is probably better since even if multiple components happen to share the same props right now, there isn't really a conceptual idea that brings them together.

also if we want to add a new prop, we'd have to split out the shared interface.

Reply:
makes sense. and the policy / agreed AI going forward is to convert class components to function components as we have to touch them, so it'll get converted sooner or later.